metrics/cgroups/oom.go

```diff
@@ -54,16 +58,26 @@ func (o *OOMCollector) Add(id, namespace string, cg cgroups.Cgroup, triggers ...
 		triggers:  triggers,
 		namespace: namespace,
 	}
-	// set the gauge's default value
-	o.memoryOOM.WithValues(id, namespace).Set(0)
 	event := unix.EpollEvent{
 		Fd:     int32(fd),
 		Events: unix.EPOLLHUP | unix.EPOLLIN | unix.EPOLLERR,
 	}
-	if err := unix.EpollCtl(o.fd, unix.EPOLL_CTL_ADD, int(fd), &event); err != nil {
-		return err
+	return unix.EpollCtl(o.fd, unix.EPOLL_CTL_ADD, int(fd), &event)
+}
+
+func (o *OOMCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- o.desc
+}
+
+func (o *OOMCollector) Collect(ch chan<- prometheus.Metric) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	for _, t := range o.set {
+		t.Lock()
+		c := t.count
+		t.Unlock()
+		ch <- prometheus.MustNewConstMetric(o.desc, prometheus.GaugeValue, float64(c), t.id, t.namespace)
```

Comment:
Should this be a counter type?

Reply:
Hum, yes, looks like it should, it never goes down.

Reply:
Well actually, if the start is recreated, it would go back down to 0